acl/acl.go

```diff
@@ -27,6 +27,11 @@ const (
 	// secret.
 	ActionPut = Action("put")
 
+	// ActionCreateVersion ("create-version" in the API) denotes permission to
+	// create a specific version of a secret if and only if that version has no
+	// current value.
```

Comment:
Optional simplification:
```suggestion
	// ActionCreateVersion ("create-version" in the API) denotes permission to
	// create a specific version of a secret if and only if that version does
	// not exist.
```

since we do not allow version values to change.

And this actually raises an interesting corner case I hadn't considered: Previously it was not possible to delete and replace a version (because any new version got a previously-unused version counter), but this allows that to happen.

I'm not sure if we care about that, but it makes me worry slightly: Does it matter if someone has a reliance interest on a value that may be now different, i.e., that the ID is no longer a reliable indicator of its content?

Reply:
Okay, I implemented this. I considered using a sentinel value, but instead I track deletions in their own map. This way we don't have to try to discern between a sentinel value and a real value.